### PR TITLE
Doc updates for use of XDP in BPF mode

### PR DESCRIPTION
**Pull request by neiljerram**

### CODE REVIEW

File: about/about-ebpf.md [hunk 1]
```diff
@@ -114,28 +115,28 @@ While the eBPF dataplane has some features that the standard Linux dataplane lac
 | Direct Server Return      | Not supported                           | Supported (requires compatible underlying network) |
```
Comment (mazdakn) [inline about/about-ebpf.md:115]: Referring to the "Throughput" factor, isn't is better to use Gbps rather than GBit. It's not part of the PR, but it got my attention! Normally, for throughput an unit of speed is needed.

File: reference/felix/configuration.md [hunk 2]
```diff
@@ -48,7 +48,6 @@ The full list of parameters which can be set is as follows.
...
 | `FelixHostname`                   | `FELIX_FELIXHOSTNAME`                   | The hostname Felix reports to the plugin. Should be used if the hostname Felix autodetects is incorrect or does not match what the plugin will expect. [Default: `socket.gethostname()`] | string |
-| `GenericXDPEnabled`               | `FELIX_GENERICXDPENABLED`               | When enabled, Felix can fallback to the non-optimized `generic` XDP mode. This should only be used for testing since it doesn't improve performance over the non-XDP mode. [Default: `false`] | boolean |
```
Comment (mazdakn) [inline reference/felix/configuration.md:51]: Just to understand, have I got it right that only 3 tables rows are re-arranged?

Comment (neiljerram) [reply reference/felix/configuration.md:51]: Yes, they are moved from the general table to the table for iptables dataplane settings.

File: reference/resources/felixconfig.md [hunk 3]
```diff
@@ -102,7 +102,7 @@ spec:
 | wireguardMTU                       | MTU set on the WireGuard interface created by Felix. Zero value means auto-detect. See [Configuring MTU]({{ site.baseurl }}/networking/mtu). | int | int | 0 |
 | wireguardRoutingRulePriority       | WireGuard routing rule priority value set up by Felix. If you change the default value, set it to a value most appropriate to routing rules for your nodes. | 1-32765 | int | 99 |
 | xdpRefreshInterval                 | Period at which Felix re-checks the XDP state in the dataplane to ensure that no other process has accidentally broken {{site.prodname}}'s rules. Set to 0 to disable XDP refresh. | `5s`, `10s`, `1m` etc. | duration | `90s` |
-| xdpEnabled                         | Enable XDP acceleration for host endpoint policies. [Default: `true`] | true,false | boolean | `true` |
+| xdpEnabled                         | When `bpfEnabled` is `false`: enable XDP acceleration for host endpoint policies.  When `bpfEnabled` is `true` this setting has no effect.  [Default: `true`] | true,false | boolean | `true` |
```
Comment (lwr20) [inline reference/resources/felixconfig.md:105]: > When `bpfEnabled` is `true` this setting has no effect

Do we need to make it explicitly clear that "no-effect" means "always on"?

File: reference/resources/felixconfig.md [hunk 4]
```diff
@@ -115,6 +115,13 @@ spec:
 | routeSource                        | Where Felix gets is routing information from for VXLAN and the BPF dataplane. The CalicoIPAM setting is more efficient because it supports route aggregation, but it only works when Calico's IPAM or host-local IPAM is in use. Use the WorkloadIPs setting if you are using Calico's VXLAN or BPF dataplane and not using Calico IPAM or host-local IPAM. | CalicoIPAM,WorkloadIPs | string | `CalicoIPAM` |
 | mtuIfacePattern                    | Pattern used to discover the host's interface for MTU auto-detection. | regex | string | `^((en|wl|ww|sl|ib)[opsx].*|(eth|wlan|wwan).*)` |
 
+<br>
+
+`genericXDPEnabled` and `xdpRefreshInterval` are only relevant when `bpfEnabled` is `false` and
+`xdpEnabled` is `true`; in other words when XDP is being used to accelerate denial-of-service
+preventation policies in the iptables dataplane.
+
+When `bpfEnabled` is `true` the "xdp" settings all have no effect.
```
Comment (lwr20) [inline reference/resources/felixconfig.md:124]: Do we need to make it explicitly clear that "no-effect" means "always on"?

File: reference/resources/felixconfig.md (hunk 3 above)
Comment (neiljerram) [reply reference/resources/felixconfig.md:105]: Well "always on" doesn't sound quite right to me.  The reality is more like a combination of
- "in BPF mode, the implementation of policy is always accelerated, using the best available BPF technology"
- "Calico v3.20 BPF mode now implements untracked ingress Deny policy, whereas previously BPF mode did not implement any untracked policy"

Can you think of wording for that that is concise enough to go here, and also precisely correct?

Comment (lwr20) [reply reference/resources/felixconfig.md:105]: Perhaps for this table we put:
```
When `bpfEnabled` is `true` this setting has no effect: in BPF mode, the implementation of policy is always accelerated, using the best available BPF technology.
```

while we put both those statements in the explanatory part of the doc (line 124ish)

Comment (neiljerram) [reply reference/resources/felixconfig.md:105]: That's nice, I'll add it.

Comment (neiljerram) [reply reference/resources/felixconfig.md:105]: Oh sorry, I didn't read your comment carefully enough.  But I don't think the statement about untracked policy being newly (and partially) supported belongs here.  And I think the shorter "has no effect" statement is better for the table.

Comment (lwr20) [reply reference/resources/felixconfig.md:105]: Perhaps:
```suggestion
| xdpEnabled                         | When `bpfEnabled` is `false`: enable XDP acceleration for host endpoint policies.  When `bpfEnabled` is `true` this setting has no effect and is treated as always `true`.  [Default: `true`] | true,false | boolean | `true` |
```

**Review by lwr20 — APPROVED** [review]
